source/extensions/filters/http/credential_injector/config.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "envoy/extensions/filters/http/credential_injector/v3/credential_injector.pb.h"
+#include "envoy/extensions/filters/http/credential_injector/v3/credential_injector.pb.validate.h"
+
+#include "source/extensions/filters/http/common/factory_base.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace CredentialInjector {
+
+class CredentialInjectorFilterFactory
+    : public Common::ExceptionFreeFactoryBase<
+          envoy::extensions::filters::http::credential_injector::v3::CredentialInjector> {
```

Comment:
nit: you can use an alias to replace all the super long proto type.

Reply:
Yah, I haven't done that cause it's a small header file and it only appears twice.

Reply:
Using the full name since:
* It's a very short header file
* It just appears twice in this file
* To avoid name conflicts

Reply:
FWIW, your code namespace will ensure the alias will not conflict with name in other namespace. But this is nit comment, it's up to you.
```
namespace Envoy {
namespace Extensions {
namespace HttpFilters {
namespace CredentialInjector {
```

Reply:
`envoy::extensions::filters::http::credential_injector::v3::CredentialInjector` and `Envoy::Extensions::InjectedCredentials::Common::CredentialInjector` would cause conflicts.